cwms_radar_api/src/main/java/cwms/radar/security/CwmsAccessManager.java

```diff
@@ -0,0 +1,204 @@
+package cwms.radar.security;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.Principal;
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+
+import cwms.radar.api.errors.RadarError;
+import io.javalin.core.security.AccessManager;
+import io.javalin.core.security.RouteRole;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jooq.DSLContext;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+
+import usace.cwms.db.jooq.codegen.packages.CWMS_ENV_PACKAGE;
+
+public class CwmsAccessManager implements AccessManager
+{
+	public static final Logger logger = Logger.getLogger(CwmsAccessManager.class.getName());
+	public static final String DATABASE = "database";
+
+	@Override
+	public void manage(@NotNull Handler handler, @NotNull Context ctx, @NotNull Set<RouteRole> requiredRoles)
+			throws Exception
+	{
+		boolean shouldProceed = isAuthorized(ctx, requiredRoles);
+
+		if(shouldProceed)
+		{
+			setSessionKey(ctx);
+
+			// Let the handler handle the request.
+			handler.handle(ctx);
+		}
+		else
+		{
+			String msg = getFailMessage(ctx, requiredRoles);
+			logger.info(msg);
+			ctx.status(HttpServletResponse.SC_UNAUTHORIZED).json(RadarError.notAuthorized());
+			ctx.status(401).result("Unauthorized");
+		}
+
+	}
+
+	private void setSessionKey(@NotNull Context ctx)
+	{
+		String sessionKey = getSessionKey(ctx);
+		if(sessionKey != null)
+		{
+			// Set the user's session key in the database.
+			Connection conn = ctx.attribute(DATABASE);
+			setSession(conn, sessionKey);
+		}
+	}
+
+	@NotNull
+	private String getFailMessage(@NotNull Context ctx, @NotNull Set<RouteRole> requiredRoles)
+	{
+		Set<RouteRole> specifiedRoles = getRoles(ctx);
+		Set<RouteRole> missing = new LinkedHashSet<>(requiredRoles);
+		missing.removeAll(specifiedRoles);
+
+		return "Request for: " + ctx.req.getRequestURI() + " denied. Missing roles: " + missing;
```

Comment:
this is technically user input, should we pass it through the sanitizer? at the least I'll make sure to check it before the next release.

Reply:
At the moment the message only goes into the log.  I don't send it back to the user.

Reply:
ah, okay. Should be fine then. At least until we get some crazy report about slf4j, anyways.